### ISSUE TITLE
(nucleus): issues with init-config flag when installing greengrass

### ISSUE THREAD

**Issue by ankitm123** [aws-greengrass/aws-greengrass-nucleus#876]
**Describe the bug**
I am using terraform to create the AWS resources required by greengrass beforehand (The ones outlined in the manual installation page: https://docs.aws.amazon.com/greengrass/v2/developerguide/manual-installation.html).
After that I use ansible to
* copy over the cert files to the root directory (`/greengrass/v2`), 
* copy over the installer zip to the GreengrassCore directory and create a config file (`~/GreengrassCore/config.yml`)
and run the command:
```
sudo -E java -Droot="{{greengrass_root_dir}}" -Dlog.store=FILE \
      -jar "{{greengrass_installer_dir}}"/lib/Greengrass.jar \
      --init-config "{{greengrass_installer_dir}}"/config.yaml \
      --component-default-user {{ggc_user}}:{{ggc_group}} \
      --setup-system-service true
```

I get a message that:
```
"Added ggc_user to ggc_group ",
"Successfully set up Nucleus as a system service"
```
However, when I check the `effectiveConfig.yaml` file, I see that the thing name, iot end point etc. are all 
empty. Checking the greengrass logs, I see the warning:
```
2021-02-26T15:08:34.687Z [WARN] (main) com.aws.greengrass.deployment.IotJobsHelper: Device not configured to talk to AWS Iot cloud. IOT job deployment is offline: [thingName cannot be empty, certificateFilePath cannot be empty, privateKeyPath cannot be empty, rootCaPath cannot be empty, iotDataEndpoint cannot be empty, iotCredEndpoint cannot be empty]. {}
```

**To Reproduce**
See above.

**Expected behavior**
Installation is successful and core device is created in the greengrass console

**Actual behavior**
Core device is not created

**Environment**
 - OS: Ubuntu 20.04
 - JDK version: `openjdk version "1.8.0_282"`
 - Nucleus version: 2.0.4

**Additional context**
If I use the provision flag, and specify the thing name and other parameters as outlined in the quick install, it does work. But creates a new certificate, and attaches it to the thing. But there are 2 issues with it:
* I have to provide AWS credentials to the installer
* Certficates are created on the fly and not in terraform state, which is a problem.

For quick install (which was successful), I see that the greengrass client was successfully created:
```
com.aws.greengrass.componentmanager.GreengrassComponentServiceClientFactory: initialize-greengrass-client. {service-region=us-east-2, service-endpoint=https://greengrass-ats.iot.us-east-2.amazonaws.com:8443}
```
So may be there are some other arguments that need to be passed for the manual step to work?

**Comment by MikeDombo** [comment]
What is the config file which you are using? It must not have setup the thing name and so on.

**Comment by ankitm123** [comment]
> What is the config file which you are using? It must not have setup the thing name and so on.
> 
> Additionally, the quotes around your template variables seems wrong. Please remove the quotes.

This is the config generated by ansible in the GreengrassCore directory;
```
---
system:
  certificateFilePath: "/greengrass/v2/device.pem.crt"
  privateKeyPath: "/greengrass/v2/private.pem.key"
  rootCaPath: "/greengrass/v2/AmazonRootCA1.pem"
  rootpath: "/greengrass/v2"
  thingName: "test-ggv2-poc"
services:
  aws.greengrass.Nucleus:
    componentType: "NUCLEUS"
    version: "2.0.4"
    configuration:
      awsRegion: "us-east-2"
      iotRoleAlias: "XXXX-role-alias"
      iotDataEndpoint: "XXXX-ats.iot.us-east-2.amazonaws.com"
      iotCredEndpoint: "XXXX.credentials.iot.us-east-2.amazonaws.com"
```

The quotes are removed once ansible evaluates the variables and prints out the config.

**Comment by MikeDombo** [comment]
What is in the effective config yaml file under the config directory in the greengrass root directory?

**Comment by ankitm123** [comment]
```
system:
  certificateFilePath: ""
  privateKeyPath: ""
  rootCaPath: ""
  rootpath: "/greengrass/v2"
  thingName: ""
services:
  aws.greengrass.Nucleus:
    componentType: "NUCLEUS"
    configuration:
      awsRegion: "us-east-2"
      componentStoreMaxSizeBytes: 10000000000
      deploymentPollingFrequencySeconds: 15
      iotCredEndpoint: ""
      iotDataEndpoint: ""
      logging: {}
      mqtt:
        spooler: {}
      networkProxy:
        proxy: {}
      platformOverride: {}
      runWithDefault:
        posixUser: "ggc_user:ggc_group"
      telemetry: {}
    dependencies: []
    version: "2.0.4"
  DeploymentService:
    dependencies: []
    version: "0.0.0"
  FleetStatusService:
    dependencies: []
  main:
    dependencies:
    - "aws.greengrass.Nucleus"
    lifecycle: {}
  TelemetryAgent:
    dependencies: []
    runtime:
      lastPeriodicAggregationMetricsTime: 1614363337396
      lastPeriodicPublishMetricsTime: 1614363337396
    version: "0.0.0"
  UpdateSystemPolicyService:
    dependencies: []
    version: "0.0.0"
```

**Comment by MikeDombo** [comment]
Please remove the quotes and ensure that the config file does exist.

It is working for me using
```
 sudo -E java -Droot=/greengrass/v2 -jar GreengrassCore/lib/Greengrass.jar --init-config t.yaml --setup-system-service true
```

Where t.yaml is 

```
 system:
    certificateFilePath: "/greengrass/v2/device.pem.crt"
    privateKeyPath: "/greengrass/v2/private.pem.key"
    rootCaPath: "/greengrass/v2/AmazonRootCA1.pem"
    rootpath: "/greengrass/v2"
    thingName: "test-ggv2-poc"
```

My logs show the line "2021-02-27T00:01:43.469Z [INFO] (main) com.aws.greengrass.config.Configuration: config-loading. Read configuration from a file path. {path=t.yaml}"

So you can see that it is reading from the t.yaml file. I see no such line in your logs which tells me that it was not able to find the file at the path that you provided. Either due to the quotes or due to the path not existing.

**Comment by MikeDombo** [comment]
I see you thumbed up my last comment; if that works for you, please let us know and close the issue.

**Comment by ankitm123** [comment]
I am trying ur suggestions, I will update the ticket once it works. Thanks for the help :)

**Comment by ankitm123** [comment]
@MikeDombo it works! The actual issue was that my config file had the yml file extension and not yaml. Once I changed it to yaml (I saw in ur example u have the yaml extension), it worked, and I can see the core device registered in the console. 

It's a bit weird that yml is not supported as an extension. What's even more weird is that I did not see that error when running remotely in ansible, but saw the error when running locally on the edge device. However, that's an ansible issue, and not greengrass.

Thanks for the tip!
